File: spring-data-requery-test-java/src/main/java/org/springframework/data/requery/domain/sample/AbstractDummy.java
package org.springframework.data.requery.domain.sample;

import io.requery.Entity;
import io.requery.Generated;
import io.requery.Key;
import io.requery.Transient;
import org.jetbrains.annotations.NotNull;
import org.springframework.data.requery.domain.AbstractPersistable;
import org.springframework.data.requery.domain.ToStringBuilder;

@Entity
public abstract class AbstractDummy extends AbstractPersistable<Integer> {

    @Key
    @Generated
    protected Integer id;

    protected String name;

    @Override
    @Transient
    protected @NotNull ToStringBuilder buildStringHelper() {
        return super.buildStringHelper()
            .add("name", name);
    }

    private static final long serialVersionUID = 1312143861898946917L;
}

File: src/test/kotlin/com/coupang/springframework/data/requery/domain/sample/AbstractDummy.kt
package com.coupang.springframework.data.requery.domain.sample

import com.coupang.springframework.data.requery.domain.AbstractPersistable
import io.requery.Column
import io.requery.Entity
import io.requery.Generated
import io.requery.Key

/**
 * com.coupang.springframework.data.requery.domain.sample.AbstractDummy
 * @author debop
 * @since 18. 5. 30
 */
@Entity
abstract class AbstractDummy @JvmOverloads constructor(aname: String? = null): AbstractPersistable<Long>() {

    @get:Key
    @get:Generated
    abstract override val id: Long?

    @get:Column(name = "dummy_name", length = 48)
    var name: String? = aname

    override fun toString(): String = "Dummy [id=$id, name=$name]"
}